rust/kernel/chrdev.rs

```diff
@@ -20,10 +19,62 @@ use crate::error::{Error, KernelResult};
 use crate::file_operations;
 use crate::types::CStr;
 
+/// Character device.
+///
+/// # Invariants
+///
+/// - [`self.0`] is valid and non-null.
+/// - [`(*self.0).ops`]   is null or has static lifetime.
+/// - [`(*self.0).owner`] is null or has static lifetime.
```

Comment:
Ugh, this isn't right. I'll update.

Reply:
Yeah, I was writing about that and then I saw the update :) Please take a look down, I think it still needs a bit of tweaking (resolving this one).